cmd/bucket_metadata_cache.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"errors"
+	"sync"
+)
+
+var errCacheEntryNotFound = errors.New("Entry not found in cache")
+
+type BucketMetadataCache struct {
+	mutex    sync.RWMutex
+	metadata map[string]BucketMetadata
+}
+
+// NewBucketMetadataCache returns a pointer to a newly created BucketMetadataCache object
+func NewBucketMetadataCache() *BucketMetadataCache {
+	return &BucketMetadataCache{
+		metadata: make(map[string]BucketMetadata),
+	}
+}
+
+// rlock locks cache for reading
+func (cache *BucketMetadataCache) rlock() {
+	cache.mutex.RLock()
+}
+
+// runlock unlocks cache locked for reading
+func (cache *BucketMetadataCache) runlock() {
+	cache.mutex.RUnlock()
+}
+
+// lock locks cache for writing
+func (cache *BucketMetadataCache) lock() {
+	cache.mutex.Lock()
+}
+
+// unlock unlocks cache locked for writing
+func (cache *BucketMetadataCache) unlock() {
+	cache.mutex.Unlock()
+}
+
+// Get fetches the given value from the cache
+func (cache *BucketMetadataCache) Get(bucket string) (BucketMetadata, error) {
+	if cache == nil {
```

Comment:
It seems that this check is redundant - if go comes here then it means that `cache` cannot be `nil`.